docker/docker/src/test/java/org/arquillian/cube/docker/impl/client/SystemPropertiesCubeSetterTest.java

```diff
@@ -0,0 +1,129 @@
+package org.arquillian.cube.docker.impl.client;
+
+import org.arquillian.cube.docker.impl.docker.DockerClientExecutor;
+import org.arquillian.cube.docker.impl.model.DockerCube;
+import org.arquillian.cube.spi.Binding;
+import org.arquillian.cube.spi.CubeRegistry;
+import org.arquillian.cube.spi.event.lifecycle.AfterDestroy;
+import org.arquillian.cube.spi.event.lifecycle.AfterStart;
+import org.jboss.arquillian.test.spi.event.suite.AfterSuite;
+import org.jboss.arquillian.test.spi.event.suite.BeforeSuite;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SystemPropertiesCubeSetterTest {
```

Comment:
you mean using Arquillian capabilities? I don't think so since it is just a fire of the events. In general I usually this approach for testing observers. I mean in this concrete simple case firing the event or calling the method directly does not offer really a plus.